packages/lesswrong/components/ultraFeed/UltraFeedSpotlightItem.tsx

```diff
@@ -0,0 +1,645 @@
+      });
+    }
+  }, [observe, spotlight]);
+
+  if (!spotlight) {
+    return null;
+  }
+
+  const url = getSpotlightUrl(spotlight);
+  const spotlightDocument = spotlight.post ?? spotlight.sequence ?? spotlight.tag;
+  const isPost = spotlight.documentType === 'Post' && !!post;
+
+  const style: React.CSSProperties & { [key: `--${string}`]: string | null } = {
+    "--spotlight-fade": spotlight.imageFadeColor,
+  }
+
+  const replyConfig = {
+    isReplying,
+    onReplyClick: () => setIsReplying(!isReplying),
+    onReplySubmit: () => setIsReplying(false),
+    onReplyCancel: () => setIsReplying(false),
+  };
+
+  return (
+    <AnalyticsContext ultraFeedElementType="feedSpotlight" spotlightId={spotlight._id} ultraFeedCardIndex={index}>
+      <div
+        ref={elementRef}
+        id={spotlight._id}
+        style={style}
+        className={classNames(classes.root, className, { [classes.rootWithFooter]: isPost, })}
+      >
+        <div className={classNames(classes.spotlightItem, {
+          [classes.spotlightFadeBackground]: !!spotlight.imageFadeColor,
+        })}>
+          <div className={classes.contentContainer}>
+            <div className={classNames(classes.content, {
+              [classes.contentWithPaddingBottom]: !isPost
+            })}>
+              <SpotlightTitle
+                spotlight={spotlight}
+                isPost={isPost}
+                url={url}
+                handleContentClick={handleContentClick}
+                className={classes.header}
+              />
+              
+              <SpotlightImage
+                spotlight={spotlight}
+                spotlightDocument={spotlightDocument}
+                className={classes.imageContainer}
+              />
+              
+              <SpotlightMetaRow
+                spotlight={spotlight}
+                spotlightDocument={spotlightDocument}
+                post={post}
+                showSubtitle={showSubtitle}
+                className={classes.metaRow}
+              />
+              
+              <div className={classes.descriptionArea}>
+                <SpotlightContentWrapper isPost={isPost} url={url} handleContentClick={handleContentClick}>
+                  <FeedContentBody
+                    html={spotlight.description?.html ?? ''}
+                    initialWordCount={SHOW_ALL_BREAKPOINT_VALUE}
+                    maxWordCount={SHOW_ALL_BREAKPOINT_VALUE}
+                    hideSuffix
+                    className={classes.description}
+                  />
+                </SpotlightContentWrapper>
+              </div>
+              
+              {isPost && post && <div className={classes.footer}>
+                  <UltraFeedItemFooter
+                    document={post}
+                    collectionName="Posts"
+                    metaInfo={{
+                      displayStatus: 'expanded',
+                      sources: ['spotlights'],
+                      lastServed: new Date(),
```

Comment:
It doesn't matter in this case because it isn't SSR'ed anyways, but in general prefer `useCurrentTime()` over `new Date()` (because the latter will change on each render)